src/metrics/winogender_metrics.py

```diff
@@ -0,0 +1,42 @@
+class GenderParity:
+    """
+  Gender parity metric from https://github.com/decompositional-semantics-initiative/DNC.
+  """
+
+    def __init__(self):
+        self.same_preds = 0.0
+        self.diff_preds = 0.0
+
+    def get_metric(self, reset=False):
+        gender_parity = float(self.same_preds) / float(self.same_preds + self.diff_preds)
+        if reset:
+            self.same_preds = 0.0
+            self.diff_preds = 0.0
+        return gender_parity
+
+    def __call__(self, predictions):
+        """
+        Calculate gender parity. 
+        Parameters
+        -------------------
+        predictions: list of dicts with fields
+            sent2_str: str, hypothesis sentence,
+            sent1_str: str, context sentence,
+            preds: int,
+            pair_id: int
+
+        Returns
+        -------------------
+        None
+        """
+        for idx in range(int(len(predictions) / 2)):
+            pred1 = predictions[idx * 2]
```

Comment:
Can you give a comment pointer to where this is being enforced? Seems like it'd be very easy to accidentally have a bug here.

Reply:
e.g. what is `len(predictions)` is odd? What if they don't line up as we're assuming here?

Reply:
batch_size needs to be even for winogender-diagnostic because predictions has to be even.